controller/src/main/java/io/pravega/controller/task/Stream/StreamTransactionMetadataTasks.java

```diff
@@ -613,40 +603,37 @@ private PingTxnStatus createStatus(Status status) {
     }
 
     public CompletableFuture<Void> writeCommitEvent(CommitEvent event) {
-        return TaskStepsRetryHelper.withRetries(() -> commitEventEventStreamWriter.writeEvent(event.getKey(), event), executor);
+        return commitWriterFuture
+                .thenCompose(commitWriter -> commitWriter.writeEvent(event.getKey(), event));
     }
 
     CompletableFuture<TxnStatus> writeCommitEvent(String scope, String stream, int epoch, UUID txnId, TxnStatus status) {
-        String key = scope + stream;
         CommitEvent event = new CommitEvent(scope, stream, epoch);
-        return TaskStepsRetryHelper.withRetries(() -> writeEvent(commitEventEventStreamWriter, commitStreamName,
-                key, event, txnId, status), executor);
+        return TaskStepsRetryHelper.withRetries(() -> writeCommitEvent(event)
+                .thenAccept(v -> {
```

Comment:
In cases where `log.isDebugEnabled()` is false (log level is set to INFO - most likely production code), you will unnecessarily execute this callback for no reason, which will eat away CPU cycles.

I suggest either adding it only if `log.isDebugEnabled() == true` or merging this and `exceptionally` below into a `whenComplete` or `handle` so you have one callback.